restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/MediaClient/AppRTCAudioManager.java

```diff
@@ -403,6 +373,38 @@ private void registerForWiredHeadsetIntentBroadcast()
          @Override
          public void onReceive(Context context, Intent intent)
          {
+
+            // Create and initialize the proximity sensor.
```

Comment:
This seems weird, how can we depend initializing stuff like setting up audio focus, etc inside the Receiver that is supposed to fire only when the a wired headset it plugged in or plugged out? Somehow this fires even when there is no headset involved, but I don't think we can trust it; seems easy to break. Isn't there a cleaner way to go?

Also, it's bad practice to create the proximity sensor each time the user plugs in and out the headset. It's lifetime should be the same as the call lifetime.

Reply:
Ok, lets talk on daily about it.